### PR TITLE
Feature/10110 add new text field with units

### PR DESCRIPTION
**Pull request by samMcCarthyBJSS**
Story [AB#10110](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/10110)
Task [AB#10339](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/10339)

### CODE REVIEW

File: app/components/input/components/text-field-with-units/macro.njk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{% macro textFieldWithUnits(params) %}
+  {%- include './template.njk' -%}
+{% endmacro %}
```
Comment (PhysicsJon) [inline app/components/input/components/text-field-with-units/macro.njk:3]: EoF newline

File: app/components/input/components/text-field-with-units/template.njk [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {% endif %}
+        <span>
+        <input class="nhsuk-input
+        {%- if params.classes %} {{ params.classes }}{% endif %}
+        {%- if params.question.error %} nhsuk-input--error{% endif %}" id="{{ params.question.id }}" name="{{ params.question.name }}" type="text"
+        {%- if params.question.data %} value="{{ params.question.data}}"{% endif %}>
+        {{params.question.unit}}</span>
+        </div>
+    </div>
+</div>
```
Comment (PhysicsJon) [inline app/components/input/components/text-field-with-units/template.njk:33]: EoF newline

**Review by PhysicsJon — DISMISSED** [review]
Just a quick look over it, will need someone with more experience to look over the rest of the code

File: app/components/input/components/text-field-with-units/macro.njk (hunk 1 above)
Comment (samMcCarthyBJSS) [reply app/components/input/components/text-field-with-units/macro.njk]: added these

File: app/components/input/components/text-field-with-units/macro.test.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  },
+};
+
+describe('textFieldWith Unit', () => {
+  it('should render the main advice', componentTester(setup, (harness) => {
+    const context = {
+      params: {
+        question: {
+          id: 'fieldId',
+          mainAdvice: 'Some really important main advice',
```
Comment (bjss-chris) [inline app/components/input/components/text-field-with-units/macro.test.js:16]: Worth moving to a constant to reuse below?

File: app/components/input/components/text-field-with-units/macro.test.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  }));
+
+  it('should render the text field with the data populated', componentTester(setup, (harness) => {
+    const context = {
+      params: {
+        question: {
+          id: 'fieldId',
+          mainAdvice: 'Some really important main advice',
+          additionalAdvice: 'Some not so important additional advice',
+          data: 'Some populated data',
```
Comment (bjss-chris) [inline app/components/input/components/text-field-with-units/macro.test.js:53]: Worth moving to a constant to reuse below?

File: app/components/input/components/text-field-with-units/macro.test.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    });
+  }));
+
+  it('should render any additional classes provided', componentTester(setup, (harness) => {
+    const context = {
+      params: {
+        question: {
+          id: 'fieldId',
+        },
+        classes: 'some-other-class',
```
Comment (bjss-chris) [inline app/components/input/components/text-field-with-units/macro.test.js:129]: Worth moving to a constant to reuse below?

File: app/components/input/components/text-field-with-units/macro.test.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  it('should render the text field as an error if the context provided contains an error', componentTester(setup, (harness) => {
+    const context = {
+      params: {
+        question: {
+          id: 'fieldId',
+          mainAdvice: 'Some really important main advice',
+          additionalAdvice: 'Some not so important additional advice',
+          data: 'Some populated data',
+          error: {
+            message: 'Some error message',
```
Comment (bjss-chris) [inline app/components/input/components/text-field-with-units/macro.test.js:73]: Worth moving to a constant and reusing with string interpolation below?

**Review by bjss-chris — COMMENTED** [review]
Just the usual questions around constant use in tests but otherwise looks good.

**Review by bjss-chris — APPROVED** [review]
👍

**Pull request merged**